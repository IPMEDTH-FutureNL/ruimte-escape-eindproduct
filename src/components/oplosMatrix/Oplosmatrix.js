import React, { useState } from 'react';
import { useNavigate } from "react-router-dom";
import Hint from '../../components/hint';
import "../../css/oplosMatrix.css"
import astronaut from'../../img/oplosmatrix/astronaut.png';
import earth from'../../img/oplosmatrix/earth.png';
import moon from'../../img/oplosmatrix/moon.png';
import lock from '../../img/oplosmatrix/lock.png';
import PopUp from '../PopUp';
import PostIt from '../../img/oplosmatrix/post-it.png';

const Oplosmatrix = () => {
    const [fout, setFout] = useState(0);
    const [text, setText] = useState("");

    const navigate = useNavigate();

    const navigation = () => {
        navigate("/puzzelDrie");
    }

    const solveLock = (e) => {
        e.preventDefault();

        const lockAnswer = 674;

        let codeString = checkInput();

        if(codeString == lockAnswer){
            setText("Je hebt de code gekraakt");
            let openpopup = document.getElementById("popup");
            openpopup.style.display = "block";
            let currentPoints = localStorage.getItem("punten");
            let pointsToInt = parseInt(currentPoints);
            pointsToInt = pointsToInt + 30;
            localStorage.setItem("punten", pointsToInt);
            navigate('/puzzelDrie');
            console.log("aantal foute pogingen " + fout);
        }else{
            setText("Dat was helaas niet goed");
            let openpopup = document.getElementById("popup");
            openpopup.style.display = "block";
            setFout(fout + 1);
        }
    }

    const checkInput = () => {
        let input = document.getElementsByClassName('oplosmatrix__lock-input');
        let lockCode = "";
        let codeString = "";
        
        for(let i = 0; i < input.length; i++){
            lockCode = input[i].value;
            codeString += lockCode.toString();
        }

        return codeString;
    }


    return(
        <section className='note'>
            <Hint type='oplosmatrixHints'/>
            <section className='note-center'>
                <div className='oplosmatrix-grid'>
                    <div className='grid-item-blank'></div>
                    <div className='grid-item-img'>
                        <img className='icon' alt="earth" src={earth} draggable="false"></img>
                    </div>
                    <div className='grid-item-img'>
                        <img className='icon' alt="moon" src={moon} draggable="false"></img>
                    </div>
                    <div className='grid-item-img'>
                        <img className='icon' alt="astronaut" src={astronaut} draggable="false"></img>
                    </div>
                    <div className='grid-item-color-red'/>
                    <div className='grid-item-number'>1</div>
                    <div className='grid-item-number'>7</div>
                    <div className='grid-item-number'>5</div>
                    <div className='grid-item-color-blue'/>
                    <div className='grid-item-number'>6</div>
                    <div className='grid-item-number'>3</div>
                    <div className='grid-item-number'>9</div>
                    <div className='grid-item-color-yellow'/>
                    <div className='grid-item-number'>2</div>
                    <div className='grid-item-number'>8</div>
                    <div className='grid-item-number'>4</div>
                </div>
            </section>
            <section className="oplosmatrix__lock">
            <img className="oplosmatrix__lock-image" src={lock} alt="oplosmatrix slot" draggable="false"/>
                <form className="oplosmatrix__form" onSubmit={(e) => solveLock(e)}>
                    <input className="oplosmatrix__lock-input" placeholder='*'></input>
                    <input className="oplosmatrix__lock-input" placeholder='*'></input>
                    <input className="oplosmatrix__lock-input" placeholder='*'></input>
                    <input className="oplosmatrix__lock-button" type="submit" value="Open" />
                </form>
            </section>
            <section className="container__postit">
                <img src={PostIt} alt="postIt" className="postIt__image" draggable="false"></img>
                <p className="postIt__text">
                        We hebben verschillende<span className="postIt__red"> planeten </span>
                        in ons zonnestelsel, 
                        op aarde zien wij bijna
                        iedere avond de <span className="postIt__blue">maan</span>. 
                        Daarbij zijn wij met een aantal
                        <span className="postIt__yellow"> astronauten </span> op het schip
                </p>
            </section>

            <PopUp text={text} onClick={() => navigation()}></PopUp>
        </section>
    )
}

export default Oplosmatrix