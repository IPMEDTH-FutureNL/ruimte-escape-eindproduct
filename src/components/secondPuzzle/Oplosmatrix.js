import React, { Component } from 'react';
import astronaut from'../../img/oplosmatrix/astronaut.png';
import earth from'../../img/oplosmatrix/earth.png';
import moon from'../../img/oplosmatrix/moon.png';
import lock from '../../img/oplosmatrix/lock.png';

const Oplosmatrix = () => {

    const solveLock = (e) => {
        e.preventDefault();

        const lockAnswer = 123;

        let codeString = checkInput();

        if(codeString == lockAnswer){
            alert("code gekraakt");
        }else{
            alert("dat was niet goed");
        }
    }

    const checkInput = () => {
        let input = document.getElementsByClassName('oplosmatrix__lock-input');
        let lockCode = "";
        let codeString = "";
        
        for(let i = 0; i < input.length; i++){
            lockCode = input[i].value;
            codeString += lockCode.toString();
        }

        return codeString;
    }


    return(
        <section className='note'>
            <section className='note-center'>
                <div className='oplosmatrix-grid'>
                    <div className='grid-item-blank'></div>
                    <div className='grid-item-img'>
                        <img className='icon' src={earth}></img>
                    </div>
                    <div className='grid-item-img'>
                        <img className='icon' src={moon}></img>
                    </div>
                    <div className='grid-item-img'>
                        <img className='icon' src={astronaut}></img>
                    </div>
                    <div className='grid-item-color-red'/>
                    <div className='grid-item-number'>1</div>
                    <div className='grid-item-number'>7</div>
                    <div className='grid-item-number'>5</div>
                    <div className='grid-item-color-blue'/>
                    <div className='grid-item-number'>6</div>
                    <div className='grid-item-number'>3</div>
                    <div className='grid-item-number'>9</div>
                    <div className='grid-item-color-yellow'/>
                    <div className='grid-item-number'>2</div>
                    <div className='grid-item-number'>8</div>
                    <div className='grid-item-number'>4</div>
                </div>
            </section>
            <section className="oplosmatrix__lock">
            <img className="oplosmatrix__lock-image" src={lock} alt="oplosmatrix slot" />
                <form className="oplosmatrix__form" onSubmit={(e) => solveLock(e)}>
                    <input className="oplosmatrix__lock-input" placeholder='*'></input>
                    <input className="oplosmatrix__lock-input" placeholder='*'></input>
                    <input className="oplosmatrix__lock-input" placeholder='*'></input>
                    <input type="submit" hidden />
                </form>
            </section>
        </section>
    )
}

export default Oplosmatrix