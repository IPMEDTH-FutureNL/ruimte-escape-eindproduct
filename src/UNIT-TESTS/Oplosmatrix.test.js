import { render , screen, fireEvent } from "@testing-library/react"
import {BrowserRouter as Router} from 'react-router-dom';
import Oplosmatrix from "../components/oplosMatrix/Oplosmatrix.js"

test("check if hints escape get rendered on start up", () => {
    const { asFragment } = render(
        <Router>
            <Oplosmatrix></Oplosmatrix>
        </Router>
    )

    screen.debug();
    expect(asFragment()).toMatchSnapshot();
})