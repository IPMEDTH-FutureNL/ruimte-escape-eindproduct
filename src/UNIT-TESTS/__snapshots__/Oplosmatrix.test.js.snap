// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if hints escape get rendered on start up 1`] = `
<DocumentFragment>
  <section
    class="note"
  >
    <button
      aria-describedby="popup-1"
      class="hint-button"
      data-testid="hintBtn"
    >
       Tips
    </button>
    <div
      class="punten"
    >
      Punten: 
    </div>
    <section
      class="note-center"
    >
      <div
        class="oplosmatrix-grid"
      >
        <div
          class="grid-item-blank"
        />
        <div
          class="grid-item-img"
        >
          <img
            alt="earth"
            class="icon"
            draggable="false"
            src="earth.png"
          />
        </div>
        <div
          class="grid-item-img"
        >
          <img
            alt="moon"
            class="icon"
            draggable="false"
            src="moon.png"
          />
        </div>
        <div
          class="grid-item-img"
        >
          <img
            alt="astronaut"
            class="icon astronautMatrix"
            draggable="false"
            src="astronaut.png"
          />
        </div>
        <div
          class="grid-item-color-red"
        />
        <div
          class="grid-item-number"
        >
          1
        </div>
        <div
          class="grid-item-number"
        >
          7
        </div>
        <div
          class="grid-item-number"
        >
          5
        </div>
        <div
          class="grid-item-color-blue"
        />
        <div
          class="grid-item-number"
        >
          6
        </div>
        <div
          class="grid-item-number"
        >
          3
        </div>
        <div
          class="grid-item-number"
        >
          9
        </div>
        <div
          class="grid-item-color-yellow"
        />
        <div
          class="grid-item-number"
        >
          2
        </div>
        <div
          class="grid-item-number"
        >
          8
        </div>
        <div
          class="grid-item-number"
        >
          4
        </div>
      </div>
    </section>
    <section
      class="oplosmatrix__lock"
    >
      <img
        alt="oplosmatrix slot"
        class="oplosmatrix__lock-image"
        draggable="false"
        src="lock.png"
      />
      <form
        class="oplosmatrix__form"
      >
        <input
          class="oplosmatrix__lock-input first"
          placeholder="*"
        />
        <input
          class="oplosmatrix__lock-input second"
          placeholder="*"
        />
        <input
          class="oplosmatrix__lock-input third"
          placeholder="*"
        />
        <input
          class="oplosmatrix__lock-button"
          type="submit"
          value="Open"
        />
      </form>
    </section>
    <section
      class="container__postit"
    >
      <img
        alt="postIt"
        class="postIt__image"
        draggable="false"
        src="post-it.png"
      />
      <p
        class="postIt__text"
      >
        We hebben verschillende
        <span
          class="postIt__red"
        >
           planeten 
        </span>
        in ons zonnestelsel, op aarde zien wij bijna iedere avond de 
        <span
          class="postIt__blue"
        >
          maan
        </span>
        . Daarbij zijn wij met een aantal
        <span
          class="postIt__yellow"
        >
           astronauten 
        </span>
         op het schip
      </p>
    </section>
    <div
      class="pop-up"
      id="popup"
    >
      <p
        class="pop-up-text"
      >
          
      </p>
      <button
        class="button primary"
        data-testid="button-popup"
      >
        Verder!
      </button>
    </div>
  </section>
</DocumentFragment>
`;
